modules/redis/redis.c

```diff
@@ -185,30 +185,33 @@ redis_dd_connect(RedisDriver *self)
 
   if (self->c->err)
     {
-      msg_error("REDIS server error, suspending",
+      msg_error("REDIS server error during connection",
                 evt_tag_str("driver", self->super.super.super.id),
                 evt_tag_str("error", self->c->errstr),
                 evt_tag_int("time_reopen", self->super.time_reopen));
-      return FALSE;
+      return LTR_NOT_CONNECTED;
     }
 
   if (self->auth)
     if (!authenticate_to_redis(self, self->auth))
       {
         msg_error("REDIS: failed to authenticate");
```

Comment:
Can you add more details to this error message?

Reply:
https://github.com/redis/hiredis/blob/master/read.h#L51
hiredis has a variety of return codes. (redisReply->type) But we are currently free-ing the reply object on the fly within `send_redis_command`.

Should I add general printout inside the send command, or change the interface of the `send_redis_command` function and delegate the ownership of the reply object to the calle?

Reply:
Tried to handle as many return value as possible. But only added it to the trace level.